include/zephyr/cache.h

```diff
@@ -93,9 +93,21 @@ extern int cache_instr_range(void *addr, size_t size, int op);
 __syscall int sys_cache_data_all(int op);
 static inline int z_impl_sys_cache_data_all(int op)
 {
-#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_DCACHE)
+#if defined(CONFIG_DCACHE)
+
+#ifdef CONFIG_CACHE_MANAGEMENT
+#if defined(CONFIG_HAS_EXTERNAL_CACHE)
 	return cache_data_all(op);
+#elif defined(CONFIG_HAS_ARCH_CACHE)
+	return arch_dcache_all(op);
+#else
+	ARG_UNUSED(op);
+
+	return -ENOTSUP;
 #endif
+#endif /* CONFIG_CACHE_MANAGEMENT */
+
+#endif /* CONFIG_DCACHE */
```

Comment:
Why? `cache_data_all` should be already pointing to `arch_dcache_all` for `CONFIG_HAS_ARCH_CACHE` and to an externally defined `cache_data_all` for `CONFIG_HAS_EXTERNA_CACHE`. This is not needed and redundant with respect to the code you have a few lines up ahead in the same file.

Reply:
my mistake